services/src/pro/contexts/mod.rs

```diff
@@ -120,6 +121,32 @@ where
         // as plots do not produce a stream of results, we have nothing to count for now
         op
     }
+
+    fn read_ml_model(&self, path: std::path::PathBuf) -> geoengine_operators::util::Result<String> {
+        let cfg = get_config_element::<crate::util::config::DataProvider>()
+            .map_err(|_| geoengine_operators::error::Error::InvalidDataProviderConfig)?;
+
+        let model_path = cfg.pro_ml_xgboost_model_path;
+
+        let model = std::fs::read_to_string(model_path.join(path))?;
```

Comment:
please use the canionicalize method to ensure the  path doesn't escape the model_path base. This should prevent bad things if `path` is something like `../../`

Reply:
https://github.com/geo-engine/geoengine/commit/5fe4b7ea1b6d3e328a3e010dc88cbddbadbb3bbb